javascript/CWE-347/MissingJWTKeyVerification.ql

```diff
@@ -0,0 +1,27 @@
+/**
+ * @name JWT missing secret or public key verification
+ * @description The application does not verify the JWT payload with a cryptographic secret or public key.
+ * @kind problem
+ * @problem.severity warning
+ * @security-severity 7.0
+ * @precision high
+ * @id js/jwt-missing-verification
+ * @tags security
+ *       external/cwe/cwe-347
+ */
+
+ import javascript
+ import DataFlow
+ import semmle.javascript.RestrictedLocations
+ 
+ from DataFlow::CallNode call, Node node, string msg
+ where
+   (call = DataFlow::moduleMember("jsonwebtoken", "verify").getACall() and
+   call.getArgument(1).analyze().getTheBooleanValue() = false
+   and node = call.getArgument(1)
+   and msg = "This argument disables the integrity enforcement of the token verification.")
+   or 
+   (call = DataFlow::moduleMember("jose", "decodeJwt").getACall() 
```

Comment:
Calling jose.decodeJwt method alone is not a fool proof detection as there is potential that the token is validated in another scenario ( jose.jwtVerify or  jose.jwtDecrypt ).  On that note should we consider dropping this into codeql repo or keep in this field pack?

Reply:
Keep in Field for now. Have you checked this against MRVA to see what % FPs this leads to? If it's significant it can't be hard to add "and not jose.jwtVerify used", right?